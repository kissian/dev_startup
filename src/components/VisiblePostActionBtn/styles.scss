.bordBody {
  border-top: 1px solid #eee;
  margin-top: 10px;
  .btn-group.actionBtnGroup {
    button.actionBtn {
      border: 0 !important;
    }
  }
}