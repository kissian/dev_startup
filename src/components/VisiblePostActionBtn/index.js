import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Col, Button, ButtonGroup } from 'react-bootstrap';
import styles from './styles.scss';

class VisiblePostActionBtn extends Component {
    render() {
        const {
            like, showComment, onToggleLike, onToggleComment,
        } = this.props;
        return (
            <Col lg={12} className={styles.bordBody}>
                <ButtonGroup className={styles.actionBtnGroup}>
                    <Button className={styles.actionBtn} onClick={onToggleLike}>
                        {
                            like ? <i className="fas fa-heart" style={{ color: 'red' }} /> : <i className="far fa-heart" />
                        }
                    </Button>
                    <Button className={styles.actionBtn} onClick={onToggleComment}>
                        {
                            showComment ? <i className="fas fa-comment" /> : <i className="far fa-comment" />
                        }
                    </Button>
                    <Button className={styles.actionBtn}>
                        <i className="far fa-share-square" />
                    </Button>
                </ButtonGroup>
            </Col>
        );
    }
}
VisiblePostActionBtn.defaultProps = {
    like: false,
    showComment: false,
    onToggleLike: () => null,
    onToggleComment: () => null,
};
VisiblePostActionBtn.propTypes = {
    like: PropTypes.bool,
    showComment: PropTypes.bool,
    onToggleLike: PropTypes.func,
    onToggleComment: PropTypes.func,
};

export default VisiblePostActionBtn;
