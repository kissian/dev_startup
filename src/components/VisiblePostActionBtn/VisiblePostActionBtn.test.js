import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import VisiblePostActionBtn from './index';

// like, showComment, onToggleLike, onToggleComment,

describe('<VisiblePostActionBtn />', () => {
  let component = null;
  let showComment = false;
  let like = false;
  const onToggleLike = jest.fn();
  const onToggleComment = jest.fn();

  it('+++VisiblePostActionBtn renders', () => {
    component = shallow(
      <VisiblePostActionBtn 
        showComment={showComment}
        like={like}
        onToggleLike={onToggleLike}
        onToggleComment={onToggleComment}
      />
    )
  })
  it('+++VisiblePostActionBtn Snapshot', () => {
    expect(component).toMatchSnapshot();
  })

  describe('+++first Modal Event Handler Test', () => {
    it('+++onToggleLike Click Event(true)', () => {
      const onToggleLikeBtn = component.find('Button').at(0);
      // console.log(onToggleLikeBtn);
      onToggleLikeBtn.simulate('click', {like: true} );
      expect(onToggleLike).toBeCalledWith({like: true});
    })

    it('+++onToggleLike Click Event(false)', () => {
      const onToggleLikeBtn = component.find('Button').at(0);
      onToggleLikeBtn.simulate('click', {like: false});
      expect(onToggleLike).toBeCalledWith({like: false});
    })

    it('+++onToggleComment Click Event(true)', () => {
      const onToggleCommentBtn = component.find('Button').at(1);
      // console.log(onToggleCommentBtn);
      onToggleCommentBtn.simulate('click', {like: true} );
      expect(onToggleComment).toBeCalledWith({like: true});
    })

    it('+++onToggleComment Click Event(false)', () => {
      const onToggleCommentBtn = component.find('Button').at(1);
      onToggleCommentBtn.simulate('click', {like: false});
      expect(onToggleComment).toBeCalledWith({like: false});
    })
  })
})