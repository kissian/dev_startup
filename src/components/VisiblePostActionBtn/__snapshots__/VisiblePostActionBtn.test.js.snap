// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VisiblePostActionBtn /> +++VisiblePostActionBtn Snapshot 1`] = `
<Col
  bsClass="col"
  className="bord-body"
  componentClass="div"
  lg={12}
>
  <ButtonGroup
    block={false}
    bsClass="btn-group"
    className="sponb-btn-group"
    justified={false}
    vertical={false}
  >
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
      onClick={[MockFunction]}
    >
      <i
        className="far fa-heart"
      />
    </Button>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
      onClick={[MockFunction]}
    >
      <i
        className="far fa-comment"
      />
    </Button>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
    >
      <i
        className="far fa-share-square"
      />
    </Button>
  </ButtonGroup>
</Col>
`;
